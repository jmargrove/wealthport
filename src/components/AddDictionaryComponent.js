import React, { Component } from "react";
import styled from "styled-components";
import CloseIcon from "mdi-react/CloseIcon";
import CheckCircleOutlineIcon from "mdi-react/CheckCircleOutlineIcon";
import { connect } from "react-redux";
import { addNewDictionary } from "../actions.js";

const mapDispatchToProps = dispatch => ({
  addNewDictionary: (name: string) => dispatch(addNewDictionary(name))
});

const AddDictionaryContainer = styled.div`
  height: 60px;
  width: 240px;
  margin: 20px;
  margin-bottom: 0;
  background-color: lightgrey;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: grab;
  &:hover {
    background-color: white;
  }
`;

const InputDictionaryContainer = styled.div`
  height: 60px;
  width: 240px;
  margin: 20px;
  margin-bottom: 0;
  background-color: white;
  display: flex;
  justify-content: center;
  align-items: center;
  overflow: auto;
`;

const ClickBox = styled.div`
  width: 25px;
  height: 25px;
  border: solid;
  border-color: black;
  border-width: thin;
  border-radius: 2px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: 5px;
  &:hover {
    background-color: lightgrey;
  }
`;

class AddDictionaryComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      toggleAddDictionaryInput: false
    };
  }

  toggleAddDictionary = () => {
    this.setState((prevState, props) => ({
      toggleAddDictionaryInput: !prevState.toggleAddDictionaryInput
    }));
  };

  handleKeyPress = e => {
    console.log(e);
    if (e.key == "Enter") {
      this.props.addNewDictionary(this.input.value);
      this.toggleAddDictionary();
    }
  };

  render() {
    console.log("the state", this.state);
    if (this.state.toggleAddDictionaryInput) {
      return (
        <InputDictionaryContainer>
          <input
            onKeyPress={this.handleKeyPress}
            autoFocus={true}
            ref={comp => (this.input = comp)}
            placeholder="Add..."
          />
          <ClickBox>
            <CloseIcon onClick={() => this.toggleAddDictionary()} />
          </ClickBox>
          <ClickBox>
            <CheckCircleOutlineIcon
              onClick={i => {
                this.props.addNewDictionary(this.input.value);
                this.toggleAddDictionary();
              }}
            />
          </ClickBox>
        </InputDictionaryContainer>
      );
    } else {
      return (
        <AddDictionaryContainer onClick={() => this.toggleAddDictionary()}>
          <div> New Dictionary </div>
        </AddDictionaryContainer>
      );
    }
  }
}

export default connect(null, mapDispatchToProps)(AddDictionaryComponent);
