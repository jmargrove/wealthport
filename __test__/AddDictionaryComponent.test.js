import React from "react";
import { shallow, mount, configure } from "enzyme";
import renderer from "react-test-renderer";
import AddDictionaryComponent from "./../src/components/AddDictionaryComponent.js";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { createStore } from "redux";
import calculatorReducers from "./../src/reducer.js";
import Adapter from "enzyme-adapter-react-15";
configure({ adapter: new Adapter() });

describe(">>>AddDictionaryComponent  +++capturing Snapshot of AddDictionaryComponent", () => {
  const mockStore = configureStore();
  let store, wrapper;

  it("+++capturing Snapshot of AddDictionaryComponent", () => {
    store = createStore(calculatorReducers);
    const renderedValue = renderer
      .create(
        <Provider store={store}>
          <AddDictionaryComponent />
        </Provider>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
