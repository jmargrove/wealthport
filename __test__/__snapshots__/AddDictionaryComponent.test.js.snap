// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>AddDictionaryComponent  +++capturing Snapshot of AddDictionaryComponent +++capturing Snapshot of AddDictionaryComponent 1`] = `
<div
  className="sc-htpNat ipTrYu"
  onClick={[Function]}
>
  <p>
     Add Dictionary 
  </p>
</div>
`;
